tutorials/notebook.md

```diff
@@ -0,0 +1,569 @@
+```javascript
+function createNotebook(creds) {
+  let client = connect(creds);
+  var keyAddr;
+
+  client.register()
+    .then(function(registerResult){
+      return client.generateKey()
+    }, function(err) {
+      console.log("Error registering wallet.");
+      console.log(err);
+    }).then(function(generateKeyResult) {
+      keyAddr = generateKeyResult.keyAddr;
+    }, function(err) {
+      console.log("Error generating key.");
+      console.log(err);
+    });
+}
+```
+
+### Creating a New Chain Tree
+Once we've created a key, we can use it to create a new chain tree to store our
+notebook data with the `createChainTree` API client method. Let's extend our
+`createNotebook` function to create a new chain tree upon successful key
+creation, handle any errors that might occur, and let's declare a new variable
+to keep track of the new chain tree's id
+
+In file `notebook/index.js`:
+```javascript
+function createNotebook(creds) {
+  let client = connect(creds);
+  var keyAddr, chainId;
+
+  client.register()
+    .then(function(registerResult){
+      return client.generateKey()
+    }, function(err) {
+      console.log("Error registering wallet.");
+      console.log(err);
+    }).then(function(generateKeyResult) {
+      keyAddr = generateKeyResult.keyAddr;
+    }, function(err) {
+      console.log("Error generating key.");
+      console.log(err);
+    }).then(function(createChainResponse) {
+      chainId = createChainResponse.chainId;
+    }, function(err) {
+      console.log("Error creating chain tree.");
+      console.log(err);
+    });
+}
+```
+
+### Saving Chain Tree Identifiers
+So now we've successfully created both a key and chain tree, and we have the
+key's public address and the chain tree's id saved in local variables to prove
+it. We still need to keep track of these identifiers between `createNotebook()`
+invocations though, so we'll need to save these somewhere more durable than a
+local variable. We could use a database for this in a full-fledged production
+app, but an external file is enough to serve our purposes.
+
+We can store the key address and the chain id in a JSON object, so let's make a
+function that creates the object for us:
+
+In file `notebook/index.js`:
+```javascript
+...
+function identifierObj(key, chain) {
+  return {
+    keyAddr: key,
+    chainId: chain
+  };
+};
+```
+
+Now that we have a way to create our identifier object, we'll need to write that
+object to a file. The `fs` module handles file i/o in Node.js, so we'll require
+it at the top of `index.js`
+
+In file `notebook/index.js`:
+```javascript
+const tupelo = require('tupelo-client');
+const fs = require('fs');
+...
+```
+
+Finally, let's pick a relative path to save the identifier file, and write a
+function to actually save it
+
+In file `notebook/index.js`:
+```javascript
+const localIdentifierPath = './.timestamper-identifiers';
+
+function dataFileExists() {
+  return fs.existsSync(localIdentifierPath);
+}
```

Comment:
I think you meant to show the code for `writeIdentifierFile` here instead of `dataFileExists` (which you also have later on right before it gets used).